app/install.php

```diff
@@ -362,9 +362,24 @@ function printStep0() {
 <?php
 }
 
+function getProcessUsername() {
+	if (function_exists('posix_getpwuid') && function_exists('posix_geteuid')) {
+		$processUser = posix_getpwuid(posix_geteuid());
+		return $processUser['name'];
+	}
+
+	exec('whoami', $output);
+	if (!empty($output)) {
```

Comment:
```suggestion
	if (!empty($output[0])) {
```

Reply:
Your suggestion will crash if `$output` is empty. On the other hand, `$output` will have some data at its first index if not empty.

Reply:
`empty($output[0])` should be save even if `$output` is null / empty / not array.
This is one of the main roles for `empty()`.
Test:

```php
<?php
$output = null;
echo empty($output[0]) ? 'empty' : 'not empty'; //no crash
```

Reply:
https://github.com/FreshRSS/FreshRSS/pull/3373/files#r563200885